### PR TITLE
feat: improve performance

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

performance 

**Did you add tests for your changes?**
existing

**If relevant, did you update the documentation?**
NA

**Summary**
- Improve performance
- inbuiltOptions as object
- unnecessary iterations purged

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

**Comment by anshumanv** [comment]
Looking at CI

**Comment by anshumanv** [comment]
Looks like some spacing diff in help output

**Comment by snitin315** [comment]
Try a rebase.

**Comment by anshumanv** [comment]
hm looks like no help

File: test/build/core-flags/optimization-flags.test.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const { run, hyphenToUpperCase } = require('../../utils/test-utils');
 const CLI = require('../../../packages/webpack-cli/lib/index');
 
 const cli = new CLI();
-const optimizationFlags = cli.getBuiltInOptions().filter(({ name }) => name.startsWith('optimization-'));
+const optimizationFlags = Object.values(cli.getBuiltInOptions().coreFlags).filter(({ name }) => name.startsWith('optimization-'));
```
Comment (evenstensberg) [inline test/build/core-flags/optimization-flags.test.js:7]: Util func?

File: packages/webpack-cli/lib/webpack-cli.js [hunk 2]
```diff
@@ -316,7 +316,7 @@ class WebpackCLI {
             return this.builtInOptionsCache;
         }
 
-        const minimumHelpFlags = [
+        const minimumHelpFlags = new Set([
```
Comment (evenstensberg) [inline packages/webpack-cli/lib/webpack-cli.js:319]: Perf here would have such little impact that there's no point. If there is 10000 objects I understand, but this is Max 30, so O(30ns) vs O(30^2) wouldnt make much difference

**Review by alexander-akait — COMMENTED** [review]
Ideally we should reduce number of `Object.entries`/`Object.values`/`Object.keys`/etc

For example we can use this for help:
```
Object.entries(minimumHelpFlags).forEach(([name, meta]) => {
  options[name].helpLevel = 'minimum';
});
```

So we avoid unnecessary looping over 1k options

File: packages/webpack-cli/lib/webpack-cli.js (hunk 2 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js:319]: we did a lookup for every flag, there could be ~300 flags

**Comment by alexander-akait** [comment]
Found more places to improve perf, I will finish it

**Comment by anshumanv** [comment]
Big thanks! 👍

**Pull request closed without merging**